index.js

```diff
@@ -205,18 +204,21 @@ export default function computeIsoline ({surface, width, height, cutoff, project
 
   // shell game time.
   // sort out shells and holes
-  HOLES: for (let hole of holes) {
+  for (let hole of holes) {
+    let foundFittingShellForHole = false
     for (let shell of shells) {
       // NB this is checking whether the first coordinate of the hole is inside the shell.
       // This is sufficient as shells don't overlap, and holes are guaranteed to be completely
       // contained by a single shell.
       if (inside(point(hole.coordinates[0][0]), shell)) {
         shell.coordinates.push(hole.coordinates)
-        continue HOLES
+        foundFittingShellForHole = true
       }
```

Comment:
In order for this to work the same as the code before, we need to add a conditional here to break the loop over shells if a hole has been found. This is why I like labels; flow control is a lot clearer if you avoid a bunch of conditionals and just say explicitly what you want to break.